Lib/fontmake/instantiator.py

```diff
@@ -224,21 +224,36 @@ def from_designspace(
         # because the math behind varLib and MutatorMath uses the default font as the
         # point of reference for all data.
         default_font = designspace.default.font
+        non_default_layer_name = designspace.default.layerName
+        default_source_uses_non_default_layer = non_default_layer_name is not None
+
         glyph_names: Set[str] = set(default_font.keys())
 
-        for source in designspace.sources:
-            other_names = set(source.font.keys())
-            diff_names = other_names - glyph_names
-            if diff_names:
-                logger.warning(
-                    "The source %s (%s) contains glyphs that are missing from the "
-                    "default source, which will be ignored: %s. If this is unintended, "
-                    "check that these glyphs have the exact same name as the "
-                    "corresponding glyphs in the default source.",
-                    source.name,
-                    source.filename,
-                    ", ".join(sorted(diff_names)),
-                )
+        if default_source_uses_non_default_layer:
```

Comment:
maybe we can remove this local variable `default_source_uses_non_default_layer` since it's only used once and simply say

```python
if non_default_layer_name is not None:
```

Reply:
Agreed, I’ve removed it. Thanks!